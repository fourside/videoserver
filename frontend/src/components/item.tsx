import * as React from 'react';
import prettyBytes from 'pretty-bytes';

interface ItemProps {
  video : Video
}

export interface Video {
  title :string
  image :string
  url :string
  category :string
  bytes :number
  mtime :string
}

export class Item extends React.Component<ItemProps, {}> {
  video :Video;

  constructor(props: ItemProps) {
    super(props);
    this.video = props.video;
  }

  render() {
    return (
      <div className="box is-half">
        <article className="media">
          <div className="media-left">
            <figure className="image is-256x256">
              <img src={this.video.image} alt={this.video.title} />
            </figure>
          </div>
          <div className="media-content">
            <div className="content">
              <p>
                <strong>{this.video.title}</strong>
              </p>
              <p>
                {tag({category: this.video.category})}
              </p>
              <p>
                <small>{this.video.mtime}</small>
              </p>
              <p>
                <span className="icon">
                  <a href={this.video.url}>
                    <i className="fas fa-download"></i>
                  </a>
                </span>
                <small>{prettyBytes(this.video.bytes)}</small>
              </p>
            </div>
          </div>
        </article>
      </div>
    );
  }
}

interface TagProps {
  category: string
}
function tag(props: TagProps) {
  let tag;
  if (props.category) {
    tag = <span className="tag is-info">{props.category}</span>;
  } else {
    tag = "";
  }
  return (
    <span>
      {tag}
    </span>
  );
}
