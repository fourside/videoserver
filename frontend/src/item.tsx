import * as React from 'react';

interface ItemProps {
  video : Video
}

export interface Video {
  title :string
  image :string
  url :string
  category :string
  bytes :number
  mtime :string
}

export class Item extends React.Component<ItemProps, {}> {
  video :Video;

  constructor(props: ItemProps) {
    super(props);
    this.video = props.video;
  }

  render() {
    return (
      <div className="box is-half">
        <article className="media">
          <div className="media-left">
            <figure className="image is-256x256">
              <img src={this.video.image} alt={this.video.title} />
            </figure>
          </div>
          <div className="media-content">
            <div className="content">
              <p>
                <strong>{this.video.title}</strong>
              </p>
              <p>
                <small>size: {this.video.bytes}</small>
              </p>
              <p>
                <small>{this.video.mtime}</small>
              </p>
            </div>
          </div>
        </article>
      </div>
    );
  }
}
