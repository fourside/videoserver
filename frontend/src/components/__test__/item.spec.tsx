import * as React from 'react';
import 'jest';
import * as renderer from 'react-test-renderer';

import { Item, Video } from '../item';

test('snapshot', () => {
  const video = {
    title: "",
    image: "",
    url: "",
    category: "",
    bytes: 256,
    mtime: "",
  };
  const component = renderer.create(<Item video={video} />).toJSON();
  expect(component).toMatchSnapshot();
});
