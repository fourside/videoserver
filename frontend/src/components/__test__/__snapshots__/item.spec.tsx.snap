// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  className="box is-half"
>
  <article
    className="media"
  >
    <div
      className="media-left"
    >
      <figure
        className="image is-256x256"
      >
        <img
          alt=""
          className="lazyload"
          data-sizes="auto"
          data-src=""
        />
      </figure>
    </div>
    <div
      className="media-content"
    >
      <div
        className="content"
      >
        <p>
          <strong>
            
          </strong>
        </p>
        <p>
          <span>
            
          </span>
        </p>
        <p>
          <small>
            
          </small>
        </p>
        <p>
          <span
            className="icon"
          >
            <a
              href=""
            >
              <i
                className="fas fa-download"
              />
            </a>
          </span>
          <small>
            256 B
          </small>
        </p>
      </div>
    </div>
  </article>
</div>
`;
